publication-commons/src/main/java/no/unit/nva/publication/model/business/Resource.java

```diff
@@ -374,5 +379,48 @@ && getStatus() == resource.getStatus()
     public Stream<TicketEntry> fetchAllTickets(ResourceService resourceService) {
         return resourceService.fetchAllTicketsForResource(this);
     }
+    
+    private static AssociatedArtifactList calculateArtifacts(Publication publication) {
```

Comment:
overloaded methods one for converting artifacts on the way in (Publication->Resource) and one on the way out (Resource->Publication)